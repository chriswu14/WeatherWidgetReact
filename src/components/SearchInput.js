import React, { Component } from "react";
import ReactDom from "react-dom";
import styled from 'styled-components';

import Flex from "./Flex";

export default class SearchInput extends Component {
    constructor() {
        super();
        this.state = {
            city: ''
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleClick = this.handleClick.bind(this);
        this.handleEnter = this.handleEnter.bind(this);
    }

    handleChange(e) {
        this.setState({
            city: e.target.value
        })
    }

    handleEnter(e) {
        if (e.keyCode === 13) {
            this.props._getWeather(this.state.city);
            this.props._getPicture(this.state.city);
        }
    }

    handleClick(e) {
        this.props._getWeather(this.state.city);
        this.props._getPicture(this.state.city);
    }

    render() {
        return (
            <Flex justifyContent="center" padding="10px">
                <StyledInput                   
                    onKeyDown={this.handleEnter}
                    onChange={this.handleChange}
                    type="text"
                    value={this.state.city}
                    placeholder="Search city..." />
                <StyledButton
                    onClick={this.handleClick}
                    disabled={!this.state.city}
                    type="button">
                    Search
                </StyledButton>
            </Flex>
        );
    }
}

const StyledInput = styled.input`
   margin-right: 10px;
   font-size: 20px;
`;

const StyledButton = styled.button`
   &:hover {
       cursor: pointer;
   }
`;
