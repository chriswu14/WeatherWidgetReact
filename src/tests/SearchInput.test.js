import React from 'react';
import SearchInput from '../components/SearchInput';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SearchInput: ', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<SearchInput />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('calls props handler with input value method after click', () => {
        const value = 'Sydney';
        const onClick = jest.fn();   
        const searchInputWrapper = mount(<SearchInput _getWeather={onClick} _getPicture={onClick} />);
        const input = searchInputWrapper.find('input');
        const button = searchInputWrapper.find('button');

        input.simulate('change', { target: { value } });
        button.simulate('click');      
        
        expect(onClick).toBeCalledWith(value);          
      });
  });