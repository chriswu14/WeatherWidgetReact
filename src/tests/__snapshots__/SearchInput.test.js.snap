// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput:  renders correctly 1`] = `
<div
  className="sc-bdVaJa eeTmpU"
>
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search city..."
    type="text"
    value=""
  />
  <button
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
